### PR TITLE
Update Developer Guide for `add_contact` feature

### PR DESCRIPTION
**Pull request by jianminglok**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -154,83 +154,62 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+    * Pros: Flexibility to add more details to the contact if needed in the future.
+    * Cons: More time required to implement.
 
-* **Alternative 2:** Individual command knows how to undo/redo by
-  itself.
-    * Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
-    * Cons: We must ensure that the implementation of each individual command are correct.
+* **Alternative 2:** Adding contact details as attributes in the `InternshipApplication` class.
+    * Pros: Easier than implement.
+    * Cons: More conflicts will occur if someone else is working on the `InternshipApplication` class at the same time.
```
Comment (jiasheng59) [inline docs/DeveloperGuide.md:212]: Good point, are you going to implement undo/redo command?

Comment (jianminglok) [reply docs/DeveloperGuide.md:212]: No, the alternatives discussed here are only about the `add_contact` feature

**Comment by laihuiqi** [comment]
Maybe we can try to style the box titles in the sequence diagrams to be more contrast from their background colors. Overall, LGTM!

**Comment by laihuiqi** [comment]
Maybe we can try to style the box titles in the sequence diagrams to be more contrast from their background colors. Overall, LGTM!

**Pull request merged**